apps/core/tasks/auto_upgrade/runner.py

```
@@ -0,0 +1,290 @@
+    if os.name == "nt" or sys.platform == "win32":
+        script = "upgrade.bat"
+    return [script, f"--{mode}"]
+
+
+def _detect_path_owner(base_dir: Path) -> tuple[str | None, str | None]:
+    """Return the owning username and home directory for ``base_dir``."""
+
+    if sys.platform == "win32":
+        return None, None
+
+    import pwd  # noqa: WPS433 - platform-specific import
```

Ruff (0.14.14)

[warning] Unused `noqa` directive (unknown: `WPS433`)

Remove unused `noqa` directive

(RUF100)